pkg/sessions/cookie/session_store.go

```diff
@@ -236,7 +236,7 @@ func joinCookies(cookies []*http.Cookie) (*http.Cookie, error) {
 	for i := 1; i < len(cookies); i++ {
 		c.Value += cookies[i].Value
 	}
-	c.Name = strings.TrimRight(c.Name, "_0")
+	c.Name = strings.TrimSuffix(c.Name, "_0")
```

Comment:
So upon a closer look, this function is quite brittle. There's a hidden requirement that the split cookie slice is passed in order.

I think this is cleaner altogether by just adding another parameter to `joinCookies` with the final merged cookie name. The `loadCookie` method that is the only place that calls this already has that as a parameter, so you can just pass it forward.

Reply:
```
func joinCookies(cookies []*http.Cookie, cookieName string) (*http.Cookie, error) {
// ...
	c.Name = cookieName
	return c, nil
}
```

Reply:
Good point. I also added test cases with arbitrary order of the split cookie slice.